src/Evolution/Systems/ScalarWave/Characteristics.hpp

```diff
@@ -0,0 +1,225 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Evolution/Systems/ScalarWave/Tags.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <typename>
+class Variables;
+
+class DataVector;
+
+namespace Tags {
+template <typename Tag>
+struct Normalized;
+}  // namespace Tags
+/// \endcond
+
+namespace ScalarWave {
+namespace Tags {
+struct ConstraintGamma2Compute : ConstraintGamma2, db::ComputeTag {
+  using argument_tags = tmpl::list<Psi>;
+  static auto function(const Scalar<DataVector>& psi) noexcept {
+    return make_with_value<type>(psi, 0.);
+  }
+  using base = ConstraintGamma2;
+};
+}  // namespace Tags
+// @{
+/*!
+ * \ingroup ScalarWave
+ * \brief Compute the characteristic speeds for the scalar wave system.
+ *
+ * Computes the speeds as described in "Optimal constraint projection for
+ * hyperbolic evolution systems" by Holst et. al \cite Holst2004wt
+ * [see text following Eq.(32)]. The characteristic fields' names used here
+ * are similar to the paper:
+ *
+ * \f{align*}
+ * \mathrm{SpECTRE} && \mathrm{Holst} \\
+ * u^{\psi} && Z^1 \\
+ * u^0_{i} && Z^{2}_{i} \\
+ * u^{\pm} && u^{1\pm}
+ * \f}
+ *
+ * The corresponding characteristic speeds \f$v\f$ are given in the text
```

Comment:
again I would suggest `\lambda_\alpha` since `v` is often a velocity in hydrodynamics.

Reply:
changed